lib/src/layer/polyline_layer/painter.dart

```diff
@@ -159,7 +159,11 @@ class _PolylinePainter<R extends Object> extends CustomPainter {
         strokeWidth = polyline.strokeWidth;
       }
 
-      final isDotted = polyline.isDotted;
+      final isDashed = polyline.pattern.segments != null;
+      final isDotted =
+          // ignore: deprecated_member_use_from_same_package
+          polyline.pattern.spacingFactor != null || polyline.isDotted;
```

Comment:
> There is no "dotted" without spacingFactor, as it defaults to 1.5 in PolylinePattern.

If I understand you correctly, I think you've made an incorrect assumption. `spacingFactor` only defaults to 1.5 when using `PolylinePattern.dotted`, otherwise it defaults to `null`.

This deprecations allows `PolylinePattern` to be `solid` (as is now default), but dotting to be enabled with a spacing factor of 1.5 when `isDotted` is set `true`.

Reply:
What I mean is that `spacingFactor` is not null if and only if we used the `dotted` constructor.
Therefore it's like
```dart
bool get isDotted => spacingFactor != null;
```

Reply:
Changed the deprecation method as you suggested, makes more sense now.